kreuzberg/_api/main.py

```
@@ -65,16 +68,118 @@ def exception_handler(request: Request[Any, Any, Any], exception: KreuzbergError
+
+    for key, value in query_params.items():
+        if value is not None and key in config_dict:
+            config_dict[key] = _convert_value_type(config_dict[key], value)
+
+    if header_config:
+        for key, value in header_config.items():
+            if key in config_dict:
+                config_dict[key] = value
+
+    if "ocr_config" in config_dict and isinstance(config_dict["ocr_config"], dict):
+        ocr_backend = config_dict.get("ocr_backend")
+        config_dict["ocr_config"] = _create_ocr_config(ocr_backend, config_dict["ocr_config"])
+
+    return ExtractionConfig(**config_dict)
+
+
 @post("/extract", operation_id="ExtractFiles")
-async def handle_files_upload(
+async def handle_files_upload(  # noqa: PLR0913
+    request: Request[Any, Any, Any],
     data: Annotated[list[UploadFile], Body(media_type=RequestEncodingType.MULTI_PART)],
+    chunk_content: str | bool | None = None,
+    max_chars: int | None = None,
+    max_overlap: int | None = None,
+    extract_tables: str | bool | None = None,
+    extract_entities: str | bool | None = None,
+    extract_keywords: str | bool | None = None,
+    keyword_count: int | None = None,
+    force_ocr: str | bool | None = None,
+    ocr_backend: Literal["tesseract", "easyocr", "paddleocr"] | None = None,
+    auto_detect_language: str | bool | None = None,
+    pdf_password: str | None = None,
 ) -> list[ExtractionResult]:
-    """Extracts text content from an uploaded file."""
-    config = try_discover_config()
+    """Extracts text content from uploaded files with optional runtime configuration.
+
+    Configuration can be provided via:
+    1. Query parameters for common settings
+    2. X-Extraction-Config header for complex nested configurations (JSON format)
+    3. Static configuration file (kreuzberg.toml or pyproject.toml)
+
+    Precedence: Header config > Query params > Static config > Defaults
+    """
+    static_config = try_discover_config()
```

we must not suppress errors stemming from file access errors (OSError due to permissions, toml parsing errors, or validation errors raised because the config is wrong) - these must bubble up as ValidationError to be processed by the exception handler.